server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -116,9 +122,10 @@ public RolloverResult rolloverClusterState(
         boolean silent,
         boolean onlyValidate,
         @Nullable IndexMetadataStats sourceIndexStats,
-        @Nullable AutoShardingResult autoShardingResult
+        @Nullable AutoShardingResult autoShardingResult,
+        FailureStoreOptions failureStoreOptions
```

Comment:
Shall we have a short trade-off discussion about using the ` FailureStoreOptions` instead of a simple `boolean`.

From a quick look, I am leaning toward a boolean because it matches better a boolean and removes the need for validation. This could always change in the future of course, and we could refactor it.

Reply:
Do you mean in this specific method, or in general for the rollover to not use `FailureStoreOptions` (i.e. make the `RolloverRequest` class have a boolean field instead of a `FailureStoreOptions` field)?

Reply:
The `RolloverRequest` already includes the `IndicesOptions` because it extends `IndicesRequest`, so I don't think that should apply there too, we would be duplicating information.

Reply:
I agree. That's also why I chose to use `FailureStoreOptions` everywhere (in parsing and in all methods), so that we have 1 single way of passing that info around. I do still get your point on changing this method to use a boolean so we don't have to have validation, though. I don't think I have a very strong opinion about either solution, do you?

Reply:
Hm, I see your point. Personally, I am leaning towards the boolean because I think the `FailureIndiceOptions` in the method `rolloverClusterState` is a bit misleading. 

Technically, this method is not handling (failure store) indices options at all, and the rollover API is not a multi-target syntax API. The `RolloverRequest` being a subclass of `IndicesOptions` is using them but they are used internally not exposed to the user. I think it makes sense to use the boolean in the `rolloverClusterState` method because we show that we do not support all failure store options but you can choose to rollover the failure store or not. 

Again, this is the rational behind my preference, feel free to leave it as is if you prefer it. I do not think having there the `FailureStoreOptions` is wrong :).

Reply:
I will throw on my agreement that `FailureStoreOptions` is a little strange to see in this method signature. I think I'd prefer to either just pass a single boolean or to pass the entire `IndicesOptions` instance in. Since we don't really need the rest of the indices options, the regular boolean seems a little simpler.